setup.py

```diff
@@ -289,23 +289,17 @@ def run(self):
 
     def build_extension(self, ext):
         name = ext.name
-        if sys.version_info[:3] >= (2, 7, 0):
```

Comment:
I thought that this is not needed because the python_requires kwarg of setuptools.setup contains ">=2.7", but it turns out that this is not checked when the setuptools.setup is called. Therefore I get the following output without my patch:

```
of macOS / OSX, or you must install from source read the documentation
here:

http://api.mongodb.org/python/current/installation.html#osx
********************************************************************

  "PyMongo supports python "
setup.py:307: UserWarning: 
********************************************************************
WARNING: The pymongo._cmessage extension module could not
be compiled. No C extensions are essential for PyMongo to run,
although they do result in significant speed improvements.
PyMongo supports python >= 2.7.

Please see the installation docs for solutions to build issues:
```
and here with my patch:
```
(oldpy26) > $ python setup.py build                                                                       [141]  [±master ✓▴]
/home/thomas/miniconda/envs/oldpy26/lib/python2.6/distutils/dist.py:266: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
running build
running build_py
running build_ext
building 'bson._cbson' extension
creating build/temp.linux-x86_64-2.6
creating build/temp.linux-x86_64-2.6/bson
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c bson/_cbsonmodule.c -o build/temp.linux-x86_64-2.6/bson/_cbsonmodule.o
In file included from /home/thomas/miniconda/envs/oldpy26/include/python2.6/Python.h:125,
                 from bson/_cbsonmodule.c:23:
/home/thomas/miniconda/envs/oldpy26/include/python2.6/modsupport.h:27:1: warning: ‘PyArg_ParseTuple’ is an unrecognized format function type [-Wformat=]
 PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
 ^~~~~~~~~~
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c bson/time64.c -o build/temp.linux-x86_64-2.6/bson/time64.o
In file included from /home/thomas/miniconda/envs/oldpy26/include/python2.6/Python.h:125,
                 from bson/time64.c:47:
/home/thomas/miniconda/envs/oldpy26/include/python2.6/modsupport.h:27:1: warning: ‘PyArg_ParseTuple’ is an unrecognized format function type [-Wformat=]
 PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
 ^~~~~~~~~~
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c bson/buffer.c -o build/temp.linux-x86_64-2.6/bson/buffer.o
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c bson/encoding_helpers.c -o build/temp.linux-x86_64-2.6/bson/encoding_helpers.o
gcc -pthread -shared build/temp.linux-x86_64-2.6/bson/_cbsonmodule.o build/temp.linux-x86_64-2.6/bson/time64.o build/temp.linux-x86_64-2.6/bson/buffer.o build/temp.linux-x86_64-2.6/bson/encoding_helpers.o -L/home/thomas/miniconda/envs/oldpy26/lib -lpython2.6 -o build/lib.linux-x86_64-2.6/bson/_cbson.so
building 'pymongo._cmessage' extension
creating build/temp.linux-x86_64-2.6/pymongo
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c pymongo/_cmessagemodule.c -o build/temp.linux-x86_64-2.6/pymongo/_cmessagemodule.o
In file included from /home/thomas/miniconda/envs/oldpy26/include/python2.6/Python.h:125,
                 from pymongo/_cmessagemodule.c:23:
/home/thomas/miniconda/envs/oldpy26/include/python2.6/modsupport.h:27:1: warning: ‘PyArg_ParseTuple’ is an unrecognized format function type [-Wformat=]
 PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
 ^~~~~~~~~~
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Ibson -I/home/thomas/miniconda/envs/oldpy26/include/python2.6 -c bson/buffer.c -o build/temp.linux-x86_64-2.6/bson/buffer.o
gcc -pthread -shared build/temp.linux-x86_64-2.6/pymongo/_cmessagemodule.o build/temp.linux-x86_64-2.6/bson/buffer.o -L/home/thomas/miniconda/envs/oldpy26/lib -lpython2.6 -o build/lib.linux-x86_64-2.6/pymongo/_cmessage.so
```
Do you think it is worth to keep this check? I mean do you really care if the output of the python setup.py build contains compiler warnings if pymongo is run with a too old python interpreter?

Reply:
I think I do prefer the old behavior because it mentions that there's a problem: "PyMongo supports python >= 2.7" and it doesn't try to compile anything. Can you please change it back?

Reply:
Part of the problem is python_requires is a fairly new concept that is ignored by a lot of old setuptools and pip versions still being used to install PyMongo.

That said, It would be nice if the warning about needing Python 2.7+ wasn't tied to compiling the C extensions. It should also be raised if you try to install from source on old Python versions with --no_ext.

Reply:
Maybe we should just have setup.py exit with an error on <2.7?

Reply:
Yes I would also prefer if we exit with an error.

See e.g. the interpreter check at the beginning of https://github.com/numpy/numpy/blob/master/setup.py#L30

Reply:
Yeah, that sounds good to me.

Reply:
Perfect thanks!